examples/DotNetCore/EventHub/EventHubService.cs

```diff
@@ -0,0 +1,156 @@
+                       .Select("Demo:Settings:*");
+
+                _refresher = options.GetRefresher();
+            });
+
+            IConfiguration configuration = builder.Build();
+        }
+
+        private void InitEventHubProcessor()
+        {
+            string eventHubConnectionString = _eventHubConnection.EventHubConnectionString;
+            string eventHubName = _eventHubConnection.EventHubName;
+            string blobStorageConnectionString = _eventHubConnection.BlobStorageConnectionString;
+            string blobContainerName = _eventHubConnection.BlobStorageContainerName;
+            string consumerGroup = _eventHubConnection.EventHubConsumerGroup;
+
+            //
+            // Check if the required configurations are available.
+            if (string.IsNullOrEmpty(eventHubConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnectionString));
+            }
+
+            if (string .IsNullOrEmpty(eventHubName))
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+
+            if (string.IsNullOrEmpty(blobStorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(blobStorageConnectionString));
+            }
+            
+            if (string.IsNullOrEmpty(blobContainerName))
+            {
+                throw new ArgumentNullException(nameof(blobContainerName));
+            }
+
+            //
+            // If no consumer group is provided, use default one.
+            if (string.IsNullOrEmpty(consumerGroup))
+            {
+                consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+            }
+
+            //
+            // Create blob storage client.
+            var storageClient = new BlobContainerClient(connectionString: blobStorageConnectionString,
+                                                        blobContainerName: blobContainerName);
+            
+            _processorClient = new EventProcessorClient(checkpointStore: storageClient,
+                                                        consumerGroup: consumerGroup,
+                                                        connectionString: eventHubConnectionString,
+                                                        eventHubName: eventHubName);
+        }
+
+        private async Task ProcessEventHandler(ProcessEventArgs eventArgs)
+        {
+            string rawEvent = Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray());
+            IEnumerable<KeyValueEvent> events = JsonConvert.DeserializeObject<List<KeyValueEvent>>(rawEvent);
+            bool shouldRefresh = false;
+
+            foreach (var e in events)
+            {
+                var key = e.Data.Key;
+                if (IsWatchedKeyValue(key))
+                {
+                    shouldRefresh = true;
+                    break;
+                }
+            }
```

Comment:
This is over complicated. We should just call `TryRefreshAsync()` if there are incoming events. If users want to scope, they can filter on the event grid side easily. No need to do this runtime checking.